File: core/src/main/java/com/andiag/core/util/StringUtils.java
package com.andiag.core.util;

public class StringUtils {

    public static String toTitleCase(String s) {
        if (s != null) {
            final String actionableDelimiters = " '-/";
            StringBuilder sb = new StringBuilder();
            boolean capNext = true;
            for (char c : s.toCharArray()) {
                c = (capNext) ? Character.toUpperCase(c) : Character.toLowerCase(c);
                sb.append(c);
                capNext = (actionableDelimiters.indexOf(c) >= 0);
            }
            return sb.toString();
        }
        return null;
    }

}

File: core/src/main/java/com/andiag/core/util/StringUtils.kt
package com.andiag.core.util

object StringUtils {

    fun toTitleCase(s: String?): String? {
        if (s != null) {
            val actionableDelimiters = " '-/"
            val sb = StringBuilder()
            var capNext = true
            for (c in s.toCharArray()) {
                sb.append(if (capNext) Character.toUpperCase(c) else Character.toLowerCase(c))
                capNext = actionableDelimiters.indexOf(c) >= 0
            }
            return sb.toString()
        }
        return null
    }

}
